packages/utils/src/ClassUtils.ts

```diff
@@ -0,0 +1,47 @@
+export type MethodName<T> = {
+  [K in keyof T]: T[K] extends (...args: unknown[]) => unknown ? K : never;
+}[keyof T];
+
+/**
+ * Bind all methods in the prototype chain to the instance (excluding those
+ * defined on Object.prototype)
+ * @param instance The instance to bind methods to
+ */
+export function bindAllMethods<T>(instance: T) {
+  const methodNames = getAllMethodNames(instance);
+
+  methodNames.forEach(methodName => {
+    // eslint-disable-next-line no-param-reassign
+    instance[methodName] = (
+      instance[methodName] as (...args: unknown[]) => unknown
+    ).bind(instance) as T[typeof methodName];
```

Comment:
Since you have to cast here anyway, I don't think the generic type is worth the added complexity